### PR TITLE
Refactor filtering in legacy API to use manager methods

### PR DESCRIPTION
**Pull request by diox**
This ensures we are always serving valid/public add-ons through the legacy API, to prepare for mixed unlisted/listed versions and the removal of the is_listed boolean in Addons.

Part of #4028

### CODE REVIEW

File: src/olympia/addons/models.py [hunk 1]
```diff
@@ -219,6 +203,71 @@ def q(*args, **kw):
...
+    def featured(self, app, lang=None, type=None):
+        """
+        Filter for all featured add-ons for an application in all locales.
+        """
+        return self.get_queryset().featured(app, lang=lang, type=type)
+
+    def listed(self, app, *status):
+        """
+        Listed add-ons have a version with a file matching ``status`` and are
+        not disabled.  Personas and self-hosted add-ons will be returned too.
```
Comment (eviljeff) [inline src/olympia/addons/models.py]: heh, you copied the ancient comment about self-hosted add-ons

File: src/olympia/legacy_api/views.py [hunk 2]
```diff
@@ -300,8 +299,8 @@ def guid_search_cache_key(guid):
         if key not in addons_xml:
             dirty_keys.add(key)
             try:
-                addon = Addon.objects.get(guid=g, disabled_by_user=False,
-                                          status__in=SEARCHABLE_STATUSES)
```
Comment (eviljeff) [inline src/olympia/legacy_api/views.py]: you can remove the `SEARCHABLE_STATUSES` constant

File: src/olympia/legacy_api/views.py [hunk 3]
```diff
@@ -300,8 +299,8 @@ def guid_search_cache_key(guid):
...
+                # Only search through public (and not disabled) add-ons.
+                addon = Addon.objects.public().get(guid=g)
```
Comment (eviljeff) [inline src/olympia/legacy_api/views.py]: one character variable names make Santa sad.

File: src/olympia/addons/models.py [hunk 4]
```diff
@@ -128,31 +128,15 @@ def clean_slug(instance, slug_field='slug'):
...
-
-    def get_queryset(self):
-        qs = super(AddonManager, self).get_queryset()
-        if not self.include_deleted:
-            qs = qs.exclude(status=amo.STATUS_DELETED)
-        if not self.include_unlisted:
-            qs = qs.exclude(is_listed=False)
-        return qs.transform(Addon.transformer)
-
+class AddonQuerySet(caching.CachingQuerySet):
```
Comment (eviljeff) [inline src/olympia/addons/models.py]: What does breaking this out give us, out of interest?  (And are there other Managers that could benefit from it?)

Comment (diox) [inline src/olympia/addons/models.py]: It's needed to allow chaining those methods, like `.public().id_or_slug(something)`. All manager methods everywhere should do it IMHO, but I went for the easy fix.

File: src/olympia/legacy_api/views.py (hunk 2 above)
Comment (diox) [inline src/olympia/legacy_api/views.py]: done

**Pull request merged**